### DISCUSSION TITLE
[Feature Request] Copy File Content button

### DISCUSSION THREAD

**Discussion by danilopolani** [community/community#4985] [Other Feature Feedback, Questions, & Ideas] [answered]
Hey,
it would be great to have a "**Copy file content**" button when we're inside a file preview, like GitLab does.
Right now to copy a file content we have to open it raw (1 click) -> select all -> copy, but I think it could be solved with a simple button.

Thank you!

**Comment by raxityo** [comment]
While GitHub considers this suggestion, this feature is available in [Refined Github](https://github.com/sindresorhus/refined-github#code) extension.

![](https://cloud.githubusercontent.com/assets/170270/14453865/8abeaefe-00c1-11e6-8718-9406cee1dc0d.png)

**Comment by davidstaheli** [comment, marked as answer]
@danilopolani, thank you for this suggestion!  All, would something like this below meet your needs?  Is there anything you'd like it to do differently?

![copy-contents](https://user-images.githubusercontent.com/1767415/130143825-6f0584d9-00e3-433d-8fab-e256a812e04b.gif)

**Reply by danilopolani** [reply]
Hey David, it looks great! This is exactly what I was thinking about, thank you 😄 

Just a personal curiosity: why there's that delay (with the spinner) between the button click and the actual code being copied in the clipboard? It's due to some internal API calls etc.? It doesn't really matter anyway, I just was making hypothesis haha.

**Reply by davidstaheli** [reply]
Great, thank you for the quick feedback! Nice observation. A 2-second sleep was temporarily added to the API just to make the different states easier to see. 😁

**Reply by danilopolani** [reply]
Oh got it, thank you for the quick answer :)

**Comment by willsmythe** [comment]
This shipped last week: https://github.blog/changelog/2021-09-20-quickly-copy-the-contents-of-a-file-to-the-clipboard/
